File: 2026/06-June/11-Thursday/solution.java
class Solution {
    private static final long MOD = 1_000_000_007L;

    private long modPow(long a, long b) {
        long res = 1;

        while (b > 0) {
            if ((b & 1) == 1)
                res = (res * a) % MOD;

            a = (a * a) % MOD;
            b >>= 1;
        }

        return res;
    }

    public int assignEdgeWeights(int[][] edges) {
        int n = edges.length + 1;

        List<Integer>[] g = new ArrayList[n + 1];
        for (int i = 0; i <= n; i++)
            g[i] = new ArrayList<>();

        for (int[] e : edges) {
            g[e[0]].add(e[1]);
            g[e[1]].add(e[0]);
        }

        Queue<Integer> q = new LinkedList<>();
        boolean[] vis = new boolean[n + 1];

        q.offer(1);
        vis[1] = true;

        int depth = -1;

        while (!q.isEmpty()) {
            int sz = q.size();
            depth++;

            while (sz-- > 0) {
                int u = q.poll();

                for (int v : g[u]) {
                    if (!vis[v]) {
                        vis[v] = true;
                        q.offer(v);
                    }
                }
            }
        }

        return (int) modPow(2, depth - 1);
    }
}

File: 2026/06-June/11-Thursday/solution.cs
public class Solution
{
    private const long MOD = 1_000_000_007;

    private long ModPow(long a, long b)
    {
        long res = 1;

        while (b > 0)
        {
            if ((b & 1) == 1)
                res = (res * a) % MOD;

            a = (a * a) % MOD;
            b >>= 1;
        }

        return res;
    }

    public int AssignEdgeWeights(int[][] edges)
    {
        int n = edges.Length + 1;

        List<int>[] g = new List<int>[n + 1];
        for (int i = 0; i <= n; i++)
            g[i] = new List<int>();

        foreach (var e in edges)
        {
            g[e[0]].Add(e[1]);
            g[e[1]].Add(e[0]);
        }

        Queue<int> q = new Queue<int>();
        bool[] vis = new bool[n + 1];

        q.Enqueue(1);
        vis[1] = true;

        int depth = -1;

        while (q.Count > 0)
        {
            int sz = q.Count;
            depth++;

            while (sz-- > 0)
            {
                int u = q.Dequeue();

                foreach (int v in g[u])
                {
                    if (!vis[v])
                    {
                        vis[v] = true;
                        q.Enqueue(v);
                    }
                }
            }
        }

        return (int)ModPow(2, depth - 1);
    }
}
